code/modules/atmospherics/environmental/LINDA_system.dm

```diff
@@ -67,18 +67,17 @@
 		var/turf/T = get_step(src, direction)
 		if(!istype(T))
 			continue
-		var/counterdir = get_dir(T, src)
 		if(CanAtmosPass(T))
 			atmos_adjacent_turfs_amount += 1
-			atmos_adjacent_turfs |= direction
-			if(!(T.atmos_adjacent_turfs & counterdir))
+			atmos_adjacent_turfs |= T
+			if(!(src in T.atmos_adjacent_turfs))

```

Comment:
There is no guarantee that the enemy turf's check atmos adjacent turfs proc will run after this one.

Thats just not how atmos works.